proposals/drafts/elections.rst

```diff
@@ -1,37 +1,249 @@
+====================
+OCDEP: Elections
+====================
+
+:Created: 2016-12-28
+:Author: James Gordon, Forest Gregg, `California Civic Data Coalition <http://www.californiacivicdata.org/>`_
+:Status: Proposed
+
+Overview
+========
+
+Definition of data types to model elections within which political contests are decided, including relationships to ballot measures and candidates competing for public office.
+
+The proposed data types are:
+
+* ``Election``
+* ``ContestBase``, a base class for:
+
+    - ``BallotMeasureContest``
+    - ``CandidateContest``
+    - ``PartyContest``
+    - ``RetentionContest``
+
+* ``Candidacy``
+* ``Party``
+* ``BallotSelectionBase``, a base class for:
+
+    - ``BallotMeasureSelection``
+    - ``CandidateSelection``
+    - ``PartySelection``
+
+Supplements the "Campaign Finance Filings" proposal being prepared by Abraham Epton.
+
+Rationale
+=========
+
+Elections are a primary focal point of civic activity in which eligible voters cast ballots to determine the outcome of political contests, including:
+
+* Who should hold a public office?
+* Should a proposed change of law be implemented?
+
+Modeling the potential outcomes of these contests is a service to voters who may cast their ballots in an impending election. Modeling the contests' actual outcomes legitimizes the election's results and enables historical electoral analysis.
+
+This proposal is submitted in response to on-going discussion around a related OCDEP focused on campaign-finance disclosures. Representing elections and their contests is necessary for modeling these disclosures because they reveal money raised and spent in support or opposition to specific candidates and ballot measures. However, since notions of elections and their contests run up against other domains, we've separated the definition of these types.
+
+Our use cases require unique representations of both previous elections and contests as well as pending elections and contests. While honoring these requirements, we also aim for consistency with the Voting Information Project's `XML format specification <http://vip-specification.readthedocs.io/en/vip5/xml/index.html#elements>`_ so as to support a high degree of interoperability with that existing data standard.
+
+VIP 5, the specification's current version, incorporates elements from the `Election Results Common Data Format Specification <https://www.nist.gov/itl/voting/nist-election-results-common-data-format-specification>`_ defined by the National Institute of Standard and Technology. As such, we have borrowed eagerly from NIST's current specification also.
+
+Important differences between this proposal and the current VIP specification are noted below.
+
+Questions
+=========
+
+* Does ``Election`` need one or two properties to store something like the election type (e.g., general, primary, special, runoff) or election level (e.g., federal, state, county, local)?
```

Comment:
I'm actually not sure we need either of these properties. Feels like this notion of "type" is already there in the election name (e.g., "2016 General") and the labels aren't mutually exclusive (e.g., aren't there special primary elections?). Also, there might be a better way to describe the geographic/jurisdictional scope of the election. It's sorta already there in `ContestBase.division_id`, but that is for individual contests.